plugin/popups/popups.py

```diff
@@ -0,0 +1,342 @@
+                            location=location,
+                            on_navigate=on_navigate)
+
+    @staticmethod
+    def cleanup_comment(raw_comment):
+        """Cleanup raw doxygen comment."""
+        def pop_prepending_empty_lines(lines):
+            first_non_empty_line_idx = 0
+            for line in lines:
+                if line == '':
+                    first_non_empty_line_idx += 1
+                else:
+                    break
+            return lines[first_non_empty_line_idx:]
+
+        import string
+        lines = raw_comment.split('\n')
+        chars_to_strip = '/' + '*' + string.whitespace
+        lines = [line.lstrip(chars_to_strip) for line in lines]
+        lines = pop_prepending_empty_lines(lines)
+        clean_lines = []
+        is_brief_comment = True
+        for line in lines:
+            if line == '' and is_brief_comment:
+                # Skip lines that belong to brief comment.
+                is_brief_comment = False
+                continue
+            if is_brief_comment:
+                continue
+            clean_lines.append(line)
+        return '\n'.join(clean_lines)
+
+    @staticmethod
+    def location_from_type(clang_type):
+        """Return location from type.
+
+        Return proper location from type.
+        Remove all inderactions like pointers etc.
+
+        Args:
+            clang_type (cindex.Type): clang type.
+
+        """
+        cursor = clang_type.get_declaration()
+        if cursor and cursor.location and cursor.location.file:
+            return cursor.location
+
+        cursor = clang_type.get_pointee().get_declaration()
+        if cursor and cursor.location and cursor.location.file:
+            return cursor.location
+
+        return None
+
+    @staticmethod
+    def link_from_location(location, text):
+        """Provide link to given cursor.
+
+        Transforms SourceLocation object into markdown string.
+
+        Args:
+            location (Cursor.location): Current location.
+            text (str): Text to be added as info.
+        """
+        result = ""
+        if location and location.file and location.file.name:
+            result += "[" + text + "]"
+            result += "(" + location.file.name
+            result += ":" + str(location.line)
+            result += ":" + str(location.column)
+            result += ") "
+        else:
+            result += text + ' '
+        return result
+
+    @staticmethod
+    def get_text_by_extent(extent):
+        """Load lines of code in range, pointed by extent.
+
+        Args:
+            extent (Cursor.extent): Ranges of source file.
+        """
+        if extent.start.file.name != extent.end.file.name:
+            return None
+
+        with open(extent.start.file.name, 'r') as f:
+            lines = f.readlines()
+            return "".join(lines[extent.start.line - 1:extent.end.line])
+
+    @staticmethod
+    def info_objc(cursor):
```

Comment:
@kjteske this is a new method for creating an info popup for objective c. I tried my best to keep the old structure and I feel like it should work out of the box. However, this requires a real world test before being pushed to everyone.

Reply:
This looks good to me after playing around with it for a few minutes. I think it's ready to push. I'll keep using it and put up a PR if I notice any problems. Thanks for keeping the obj-c support!

Reply:
I'll see if I can add useful tests to test_errors_vis.py this weekend.

Reply:
No rush. If this works for now take your time. I just want to be sure I can keep supporting this without bugging you every time I need to touch this code.